### PR TITLE
[DET-3026] login flake

### PR DESCRIPTION
**Pull request by hamidzr**
Sign inout -- should be able to log in (failed)

# Test Plan
- yes

### CODE REVIEW

**Comment by hamidzr** [comment]
TODO find and treat other instances of UI triggered requests that could into get into this condition

**Comment by hamidzr** [comment]
e2e tests passed. gonna restart them a few times

File: webui/tests/cypress/integration/02-loginout.spec.ts [hunk 1]
```diff
@@ -25,7 +25,10 @@ describe('Sign in/out', () => {
     cy.visit('/ui');
     cy.get('#avatar').click();
     cy.get(`nav a[href="${LOGOUT_ROUTE}"]`).should('have.lengthOf', 1);
+    cy.server();
+    cy.route('POST', '/logout').as('logoutRequest');
```
Comment (hkang1) [inline webui/tests/cypress/integration/02-loginout.spec.ts:29]: question: does `cy.route` double up on sending the `/logout` POST request which the `Sign Out` button performs as well? If that's the case, if for whatever reason the signout button fails to send a `/logout` request down the line, will this test pass then?

Comment (hkang1) [reply webui/tests/cypress/integration/02-loginout.spec.ts:29]: nm, cypress docs were not clear, but found the answer on the good ole' stackoverflow. So it looks like it's a xhr listener essentially and the wait command after the click will wait until it sees the logout POST.

Comment (hamidzr) [reply webui/tests/cypress/integration/02-loginout.spec.ts:29]: https://docs.cypress.io/api/commands/as.html#DOM-element is the doc I went off of

**Comment by hamidzr** [comment]
saw a flake 30 minutes ago https://app.circleci.com/pipelines/github/determined-ai/determined/1031/workflows/74429cc6-9966-4627-a64d-e1cd99bd3c34/jobs/16222/steps

**Comment by hkang1** [comment]
> saw a flake 30 minutes ago https://app.circleci.com/pipelines/github/determined-ai/determined/1031/workflows/74429cc6-9966-4627-a64d-e1cd99bd3c34/jobs/16222/steps

This flake revolves around `/login` getting canceled and not that we're not waiting properly on the endpoint response. Any clues as to what causes the canceling?

**Comment by hkang1** [comment]
https://docs.cypress.io/guides/references/changelog.html#3-3-0
> There’s a new ‘canceled’ XHR status shown on the Command Log to indicate the difference between calling xhr.abort() and the browser canceling the request due to page navigation. Addresses #3973.

the `canceled` xhr status seems to indicate there is a browser navigating away from the page after the request has been sent.

additional example:
https://github.com/cypress-io/cypress/issues/3973#issuecomment-483559122

**Pull request merged**